shell/platform/fuchsia/flutter/text_delegate_unittests.cc

```diff
@@ -0,0 +1,286 @@
+// it was called with.
+class FakeImeService : public fuchsia::ui::input::ImeService {
+ public:
+  virtual void GetInputMethodEditor(
+      fuchsia::ui::input::KeyboardType keyboard_type,
+      fuchsia::ui::input::InputMethodAction action,
+      fuchsia::ui::input::TextInputState input_state,
+      fidl::InterfaceHandle<fuchsia::ui::input::InputMethodEditorClient> client,
+      fidl::InterfaceRequest<fuchsia::ui::input::InputMethodEditor> ime) {
+    keyboard_type_ = std::move(keyboard_type);
+    action_ = std::move(action);
+    input_state_ = std::move(input_state);
+    client_ = client.Bind();
+    ime_ = std::move(ime);
+  }
+
+  virtual void ShowKeyboard() { keyboard_shown_ = true; }
+
+  virtual void HideKeyboard() { keyboard_shown_ = false; }
+
+  bool IsKeyboardShown() { return keyboard_shown_; }
+
+  bool keyboard_shown_ = false;
+
+  fuchsia::ui::input::KeyboardType keyboard_type_;
+  fuchsia::ui::input::InputMethodAction action_;
+  fuchsia::ui::input::TextInputState input_state_;
+  fidl::InterfacePtr<fuchsia::ui::input::InputMethodEditorClient> client_;
+  fidl::InterfaceRequest<fuchsia::ui::input::InputMethodEditor> ime_;
+};
+
+class TextDelegateTest : public ::testing::Test {
+ protected:
+  TextDelegateTest()
+      : loop_(&kAsyncLoopConfigAttachToCurrentThread),
+        keyboard_service_binding_(&keyboard_service_),
+        ime_service_binding_(&ime_service_) {
+    fidl::InterfaceHandle<fuchsia::ui::input3::Keyboard> keyboard_handle;
+    auto keyboard_request = keyboard_handle.NewRequest();
+    keyboard_service_binding_.Bind(keyboard_request.TakeChannel());
+
+    fidl::InterfaceHandle<fuchsia::ui::input::ImeService> ime_service_handle;
+    ime_service_binding_.Bind(ime_service_handle.NewRequest().TakeChannel());
+
+    auto fake_view_ref_pair = scenic::ViewRefPair::New();
+
+    text_delegate_ = std::make_unique<TextDelegate>(
+        std::move(fake_view_ref_pair.view_ref), std::move(ime_service_handle),
+        std::move(keyboard_handle),
+        // Should this be accessed through a weak pointer?
+        [this](std::unique_ptr<flutter::PlatformMessage> message) {
+          last_message_ = std::move(message);
+        });
+
+    // TextDelegate has some async initialization that needs to happen.
+    RunLoopUntilIdle();
+  }
+
+  // Runs the event loop until all scheduled events are spent.
+  void RunLoopUntilIdle() { loop_.RunUntilIdle(); }
+
+  void TearDown() override {
+    loop_.Quit();
+    ASSERT_EQ(loop_.ResetQuit(), 0);
+  }
+
+  async::Loop loop_;
+
+  FakeKeyboardService keyboard_service_;
+  fidl::Binding<fuchsia::ui::input3::Keyboard> keyboard_service_binding_;
+
+  FakeImeService ime_service_;
+  fidl::Binding<fuchsia::ui::input::ImeService> ime_service_binding_;
+
+  // Unit under test.
+  std::unique_ptr<TextDelegate> text_delegate_;
+
+  std::unique_ptr<flutter::PlatformMessage> last_message_;
+};
+
+TEST_F(TextDelegateTest, ActivateIme) {
```

Comment:
For unit tests sometimes it's hard to tell what they're doing from the name alone, so it can help readability to add a quick summary of what the unit test is focusing on as a comment at the top of the test.

Reply:
Point taken. Done.